tests/testthat/test-tab_afun_cfun.R

```diff
@@ -0,0 +1,256 @@
+context("Analysis functions (afun)")
+
+test_that(".spl_context contains information about the column split", {
+    ## Duplication hack # xxx send it to Gabe
+    DM_tmp <- DM %>% 
+        mutate(method = factor("Mean")) 
+    
+    DM_tmp <- rbind(DM_tmp, DM_tmp %>% 
+                        mutate(method = factor("SD")))
+    
+    analysis_fun_fin <- function(x, .spl_context, labelstr = "", ...) {
+        
+        # Very smart internal checks for name reconstruction from path
+        stopifnot(length(.spl_context$cur_col_id[[1]]) == 1L)
+        stopifnot(.spl_context$cur_col_id[[1]] %in% names(.spl_context))
+        
+        if (any(.spl_context$cur_col_split_val[[2]] == "SD")) {
+            res <- list("SOMETHING" = sd(x))
+        } else if (any(.spl_context$cur_col_split_val[[2]] == "Mean")) {
+            res <- list("SOMETHING" = mean(x))
+        }
+            
+        in_rows(.list = res)
+    }
+    
+    lyt <- basic_table() %>% 
+        split_rows_by("STRATA1") %>%
+        split_cols_by(var = "method") %>%
+        split_cols_by("SEX", split_fun = drop_split_levels) %>%
+        analyze(vars = "BMRKR1", afun = analysis_fun_fin, format = "xx.xxx")
+    
+    expect_silent(tbl <- lyt %>% build_table(DM_tmp))
+    
+    tol <- 0.02
+    DM_tmp_F <- DM_tmp %>% filter(SEX == "F", STRATA1 == "B")
+    expect_equal(tbl[4, 1, drop = TRUE], mean(DM_tmp_F$BMRKR1), tolerance = tol)
+    expect_equal(tbl[4, 3, drop = TRUE], sd(DM_tmp_F$BMRKR1), tolerance = tol) # xxx why -0.011?
```

Comment:
```suggestion
    DM_B_F <- DM %>% filter(SEX == "F", STRATA1 == "B")
    expect_equal(tbl[4, 1, drop = TRUE], mean(DM_B_F$BMRKR1))
    expect_equal(tbl[4, 3, drop = TRUE], sd(DM_B_F$BMRKR1))
```

Reply:
Either need to filter original DM before calculating stats or need to subset by `method`. Doesn't make a difference for mean() but it affects sd().